src/hera/workflows/runner.py

```diff
@@ -101,11 +119,13 @@ def _runner(entrypoint: str, kwargs_list: Any) -> str:
         if "name" not in kwarg or "value" not in kwarg:
             continue
         # sanitize the key for python
-        key = cast(str, serialize(kwarg["name"])).replace("-", "_")
+        key = cast(str, serialize(kwarg["name"]))
```

Comment:
We've lost the `.replace("-", "_")` behaviour for non-experimental users. We could either do it here or in the `_map_keys` function?

I'm not sure whether the script runner will pick up a global config defined next to the given function in the module though, might need an extra parameter or alternative runner 🤔

Reply:
I think it's OK to break in this case as it's a experimental feature.

Reply:
Actually I see the suggestion you have made. That makes more sense.